jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -1862,6 +1905,13 @@ ecma_compare_ecma_non_direct_strings (const ecma_string_t *string1_p, /**< ecma-
     return false;
   }
 
+#ifndef CONFIG_DISABLE_ES2015_MAP_BUILTIN
+  if (string1_container == ECMA_STRING_CONTAINER_MAP_KEY)
+  {
+    return ecma_op_same_value_zero (string1_p->u.value, string2_p->u.value);
```

Comment:
This looks like a quite costly operation.

Reply:
Indeed. It's a bit costly but it's required, since with the current implementation the following tests are failing.

```js
var m = new Map();
m.set(NaN, "not a number");
assert(m.size === 1);
assert(m.get(NaN) === "not a number"); // Currently: m.get(NaN) is undefined
assert(m.get(Number("foo")) === "not a number") // Currently: m.get(Number("foo") is undefined

m.set(0, "PosZero");
assert(m.size === 2); // Currently: m.size === 2
m.set(-0, "NegZero");
assert(m.size === 2); // Currently: m.size === 3
assert(m.get (0) === "NegZero");
assert(m.get (-0) === "NegZero");

```